monkey/monkey_island/cc/ui/src/components/Main.tsx

```diff
@@ -42,19 +44,36 @@ export const IslandRoutes = {
   PluginsPage: '/marketplace',
 }
 
-export function isReportRoute(route){
+export function isReportRoute(route) {
   return route.startsWith(IslandRoutes.Report);
 }
 
 class AppComponent extends AuthComponent {
   private interval: Timeout;
+  private fetched: boolean;
 
   constructor(props) {
     super(props);
     this.state = {
       completedSteps: new CompletedSteps(false),
+      availablePlugins: [],
+      installedPlugins: [],
+      numberOfPluginsThatRequiresUpdate: 9
     };
     this.interval = undefined;
+    this.fetched = false;
+  }
+
+  setAvailablePlugins = (plugins) => {
+    this.setState({
+        availablePlugins: plugins
+    });
+  }
+
+  setInstalledPlugins = (plugins) => {
```

Comment:
I don't see where `installedPlugins` is being updated. This is a reduction in functionality.